### PR TITLE
docs: auto-generate list of CLI commands and insert into docs/warcli.md

### PR DESCRIPTION
**Pull request by pinheadmz**
uses tabulate assuming merge of #201 

check it out: https://github.com/pinheadmz/warnet/blob/api-docs/docs/warcli.md

### CODE REVIEW

**Comment by willcl-ark** [comment]
Looks awesome to me! Nice.

I'll fix up #201 first then we can do this one?

**Comment by pinheadmz** [comment]
> I'll fix up #201 first then we can do this one?

cool, good plan. might tweak the style here and there on this as well

**Comment by willcl-ark** [comment]
Code LGTM in general.

Just gave this a quick whizz rebase on top of #211 and it seemed to work well, except it uses my graph file path which needs manual change

image

It could be fixed with a patch like this, I believe:

```patch
diff --git a/src/warnet/cli/network.py b/src/warnet/cli/network.py
index 87f2d97..b235d3f 100644
--- a/src/warnet/cli/network.py
+++ b/src/warnet/cli/network.py
@@ -1,5 +1,5 @@
 import base64
-from pathlib import Path
+from pathlib import Path, PurePath
 
 import click
 from rich import print
@@ -8,7 +8,7 @@ from graphs import GRAPHS
 from warnet.cli.rpc import rpc_call
 
 
-DEFAULT_GRAPH_FILE = GRAPHS / "default.graphml"
+DEFAULT_GRAPH_FILE = PurePath(GRAPHS / "default.graphml")
 
 
 @click.group(name="network")
@@ -17,7 +17,7 @@ def network():
 
 
 @network.command()
-@click.argument("graph_file", default=DEFAULT_GRAPH_FILE, type=click.Path())
+@click.argument("graph_file", default=DEFAULT_GRAPH_FILE.relative_to(Path.cwd()), type=click.Path())
 @click.option("--force", default=False, is_flag=True, type=bool)
 @click.option("--network", default="warnet", show_default=True)
 def start(graph_file: Path, force: bool, network: str):
```

...which would use the relative path from where the script was run, to the graph file, instead.

**Comment by pinheadmz** [comment]
thanks for reviewing Will, I modified the script to convert paths to relative when needed: 17da8a4b077a527a635f4656ea0b3db716efee5f

File: scripts/apidocs.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    global doc
+    doc += f"### `warcli{super} {cmd['name']}`" + "\n"
+    doc += cmd["help"].strip().replace("<", "\\<") + "\n"
+    doc += "\noptions:\n"
+    headers = ["name", "type", "required", "default"]
+    data = [[
+        p['name'],
+        p['type']['param_type'],
+        p['required'],
+        p['default'] if p['type']['param_type'] != "Path" else Path(p['default']).relative_to(Path.cwd())
```
Comment (pinheadmz) [inline scripts/apidocs.py:20]: only caveat: script must be run from root directory. im ok with that

Comment (willcl-ark) [reply scripts/apidocs.py:20]: I think that's fine too

**Review by willcl-ark — APPROVED** [review]
ACK 17da8a4b077a527a635f4656ea0b3db716efee5f

Re-ran after new relative path changes and got a 0 line diff 👍🏼 

```
will@ubuntu in ~/src/warnet on  api-docs [$?] is 📦 v1.0.0 via  v18.16.0 : 🐍 warnet
₿ python scripts/apidocs.py

will@ubuntu in ~/src/warnet on  api-docs [$?] is 📦 v1.0.0 via  v18.16.0 : 🐍 warnet
₿ git diff
```

**Pull request merged**